api/serve.py

```
@@ -128,6 +131,64 @@ def _generate_smart_sql(question: str, schema: str) -> tuple[str, float, float]:
                 filter_col = col
             break
 
+    def _extract_where_clause(q: str, text_cols: list, columns: list) -> str | None:
```

Ruff (0.15.6)

[warning] Too many branches (14 > 12)

(PLR0912)